### PR TITLE
Lucianoeli/content provider

### PR DESCRIPTION
**Pull request by lucianoeli**
Retrieving data from local realm repository

### CODE REVIEW

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/local/SaveCharactersLocalImpl.kt [hunk 1]
```diff
@@ -11,7 +11,5 @@ class SaveCharactersLocalImpl(private val mapper: CharacterMapperLocal = Charact
         realm.beginTransaction()
```
Comment (priettt) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/local/SaveCharactersLocalImpl.kt]: Try to use executeTransaction(), as it is less error prone.

Comment (lucianoeli) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/local/SaveCharactersLocalImpl.kt]: done

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -3,4 +3,6 @@
     <string name="message_no_items_to_show">No items to show</string>
     <string name="no_description_available">No description available</string>
     <string name="img_content_description">Image content description</string>
+    <string name="CharacterDetailDialog">CharacterDetailDialog</string>
```
Comment (priettt) [inline app/src/main/res/values/strings.xml]: I think it'd better to use the same naming_convention for all the strings in the project

Comment (SartiMau) [reply app/src/main/res/values/strings.xml]: +1

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/CharacterProvider.kt [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    init {
+        sURIMatcher.addURI(AUTHORITY, CHARACTERS_TABLE, CHARACTERS)
+    }
+
+    override fun onCreate(): Boolean {
+        try {
+            Realm.init(context)
+            val config = RealmConfiguration.Builder()
+                    .name("realm.character")
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/CharacterProvider.kt]: use constants for this pls

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/CharacterProvider.kt [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                            char.thumbnail?.path,
+                            char.thumbnail?.extension
+                    ))
+        }
+        cursor.setNotificationUri(context.contentResolver, CONTENT_URI)
+        return cursor
+    }
+
+    override fun insert(uri: Uri?, values: ContentValues?): Uri {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/CharacterProvider.kt]: Why do you have these methods unimplemented?

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/CharacterProvider.kt (hunk 4 above)
Comment (lucianoeli) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/CharacterProvider.kt]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/CharacterProvider.kt (hunk 3 above)
Comment (lucianoeli) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/CharacterProvider.kt]: done

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (lucianoeli) [reply app/src/main/res/values/strings.xml]: done

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/CharacterProvider.kt [hunk 5]
```diff
@@ -74,18 +79,41 @@ class CharacterProvider : ContentProvider() {
...
+                    )
+            )
+        }
+
+        val realm = Realm.getDefaultInstance()
+        realm.executeTransaction {
+            it.insertOrUpdate(characterRealm as RealmModel)
+        }
+
+        return Uri.parse(CHARACTERS_TABLE + "/" +characterRealm?.id)
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/CharacterProvider.kt]: use kotlin concat string pls

**Review by SartiMau — APPROVED** [review]

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -3,4 +3,6 @@
     <string name="message_no_items_to_show">No items to show</string>
     <string name="no_description_available">No description available</string>
     <string name="img_content_description">Image content description</string>
+    <string name="title_character_detail_fragment">CharacterDetailDialog</string>
+    <string name="toastMsg_data_loaded_from_local_repo">Data loaded from local repository</string>
```
Comment (priettt) [inline app/src/main/res/values/strings.xml]: This should be toast_msg_dat... etc.

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/CharacterProvider.kt (hunk 5 above)
Comment (lucianoeli) [reply app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/CharacterProvider.kt]: done

File: app/src/main/res/values/strings.xml (hunk 6 above)
Comment (lucianoeli) [reply app/src/main/res/values/strings.xml]: done

**Review by priettt — APPROVED** [review]

File: app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/ContentLoader.kt [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+import com.puzzlebench.clean_marvel_kotlin.data.provider.CharacterProvider.Companion.ID_COLUM
+import com.puzzlebench.clean_marvel_kotlin.data.provider.CharacterProvider.Companion.NAME_COLUM
+import com.puzzlebench.clean_marvel_kotlin.data.provider.CharacterProvider.Companion.THUMBNAIL_EXTENSION_COLUM
+import com.puzzlebench.clean_marvel_kotlin.data.provider.CharacterProvider.Companion.THUMBNAIL_PATH_COLUM
+import com.puzzlebench.clean_marvel_kotlin.domain.model.Character
+import com.puzzlebench.clean_marvel_kotlin.domain.model.Thumbnail
+import com.puzzlebench.clean_marvel_kotlin.presentation.MainActivity
+import com.puzzlebench.clean_marvel_kotlin.presentation.mvp.view.CharacterView
+
+class ContentLoader(val context: MainActivity, val view: CharacterView) : LoaderManager.LoaderCallbacks<Cursor> {
```
Comment (fanky10) [inline app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/provider/ContentLoader.kt:19]: this is a loader specifically for Characters, I would rename it to: CharactersLoader or similar

**Review by fanky10 — COMMENTED** [review]
minor comment otherwise it's good to go

**Comment by fanky10** [comment]
Looks good to me

**Pull request merged**